File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams
{
  gender?:string;
  minAge = 0;
  maxAge = 77;
  pageNumber = 1;
  pageSize = 5;
  userType:string;
  orderBy ='lastActive'
  constructor(user: User)
  {
    this.gender = 'female';
    // this.gender = user.gender === 'female'? 'male': 'female';
    this.userType = 'Patient';
  }
}

File: API/Helpers/UserParams.cs
using API.Entities;

namespace API.Helpers;

public class UserParams
{
    private const int MaxPageSize = 50;
    private int _pageSize = 10;
    public int PageNumber { get; set; } = 1;

    public int PageSize
    {
        get => _pageSize;
        set => _pageSize = value > MaxPageSize ? MaxPageSize : value;
    }

    public string CurrentUsername { get; set; }
    public string? Gender { get; set; }
    public UserTypes UserType { get; set; }
    public int MinAge { get; set; } = 0;
    public int MaxAge { get; set; } = 150;

    public string OrderBy { get; set; } = "lastActive";
}